src/ecs/saveload/marker.rs

```diff
@@ -0,0 +1,122 @@
+//! Provides with `Marker` and `MarkerAllocator` traits
+
+use std::hash::Hash;
+use std::collections::HashMap;
+
+use shred::Resource;
+use specs::{Component, Entities, Entity, DenseVecStorage, Join, ReadStorage, WriteStorage};
+
+use super::SerializableComponent;
+
+
+/// This trait should be implemetened by component which is gonna be used as marker
+/// This marker should be set to entity that should be serialized
+/// If serialization strategy needs to set marker to some entity it should use new marker allocated for `Marker::Allocator`
+pub trait Marker: SerializableComponent + Copy {
+    /// Id of the marker
+    type Identifier: Copy + ::std::fmt::Debug + Eq + Hash;
+
+    /// Allocator for this `Marker`
+    type Allocator: MarkerAllocator<Self>;
+
+    /// Get this marker id
+    fn id(&self) -> Self::Identifier;
+
+    /// Update marker with new value
+    fn update(&mut self, _other: Self) {}
+}
+
+/// This allocator is used with `Marker` trait
```

Comment:
Might be nice to provide a small example of implementing this trait if we are gonna let people implement it themselves.

Reply:
`NetMarker` will implement this function. Maybe I can add it as an example into doc